packages/environment-check/src/checks/helpers/service-checks.ts

```diff
@@ -0,0 +1,92 @@
+import type { AbapServiceProvider } from '@sap-ux/axios-extension';
+import type { ResultMessage } from '../../types';
+import { TransportChecksService } from '@sap-ux/axios-extension';
+import { getLogger } from '../../logger';
+import { t } from '../../i18n';
+
+/**
+ * Checks if ATO catalog is available.
+ *
+ * @param provider the AbapServiceProvider to be used
+ * @returns Result messages and result of ato catalog request
+ */
+export async function checkAtoCatalog(
+    provider: AbapServiceProvider
+): Promise<{ messages: ResultMessage[]; isAtoCatalog: boolean }> {
+    const logger = getLogger();
+    let isAtoCatalog = false;
+    try {
+        const atoInfo = await provider.getAtoInfo();
```

Comment:
It is better to call `provider.getAdtService` directly (i.e. the implementation of getAtoInfo()). Because you want to validate if the service is null or not. Current getAtoInfo() implementation already hides this check. You don't know if atoInfo is returning empty object because its returned by backend or because the service is null.

Reply:
Updated